apps/api-extractor-model/src/model/ApiTypeParameter.ts

```diff
@@ -0,0 +1,109 @@
+import { ApiDeclaredItem, IApiDeclaredItemOptions, IApiDeclaredItemJson } from '../items/ApiDeclaredItem';
+import { ApiItemKind } from '../items/ApiItem';
+
+/**
+ * Constructor options for {@link TypeParameter}.
+ * @public
+ */
+export interface IApiTypeParameterOptions extends
+  IApiNameMixinOptions,
+  IApiDeclaredItemOptions {
+
+  constraintTokenRange: IExcerptTokenRange | undefined;
+  defaultTypeTokenRange: IExcerptTokenRange | undefined;
+}
+
+export interface IApiTypeParameterJson extends
+  IApiNameMixinJson,
+  IApiDeclaredItemJson {
+
+  constraintTokenRange?: IExcerptTokenRange;
+  defaultTypeTokenRange?: IExcerptTokenRange;
+}
+
+/**
+ * Represents a named type parameter for a generic declaration.
+ *
+ * @remarks
+ *
+ * `TypeParameter` represents a TypeScript declaration such as `T` in this example:
+ *
+ * ```ts
+ * export interface Array<T> {
+ * }
+ * ```
+ *
+ * `TypeParameter` objects belong to the {@link (ApiTypeParameterListMixin:interface).typeParameters} collection.
+ *
+ * @public
+ */
+export class ApiTypeParameter extends ApiNameMixin(ApiDeclaredItem) {
+  /**
+   * An {@link Excerpt} that describes the base constraint of the type parameter.
+   */
+  public readonly constraintExcerpt: Excerpt | undefined;
+  /**
+   * An {@link Excerpt} that describes the default type of the type parameter.
+   */
+  public readonly defaultTypeExcerpt: Excerpt | undefined;
+
+  /** @override */
+  public static onDeserializeInto(options: Partial<IApiTypeParameterOptions>, jsonObject: IApiTypeParameterJson): void {
+    super.onDeserializeInto(options, jsonObject);
+
+    options.constraintTokenRange = jsonObject.constraintTokenRange;
+  }
+
+  public static getCanonicalReference(name: string): string {
+    return `(${name}:typeparam)`;
+  }
+
+  public constructor(options: IApiTypeParameterOptions) {
+    super(options);
+    this.constraintExcerpt = options.constraintTokenRange &&
+      this.buildExcerpt(options.constraintTokenRange);
+    this.defaultTypeExcerpt = options.defaultTypeTokenRange &&
+      this.buildExcerpt(options.defaultTypeTokenRange);
+  }
+
+  /** @override */
+  public get kind(): ApiItemKind {
+    return ApiItemKind.TypeParameter;
+  }
+
+  /** @override */
+  public get canonicalReference(): string {
+    return ApiTypeParameter.getCanonicalReference(this.name);
+  }
+
+  /** @override */
+  public serializeInto(jsonObject: Partial<IApiTypeParameterJson>): void {
+    super.serializeInto(jsonObject);
+
+    jsonObject.constraintTokenRange = this.constraintExcerpt &&
+      this.constraintExcerpt.tokenRange;
+    jsonObject.defaultTypeTokenRange = this.defaultTypeExcerpt &&
+      this.defaultTypeExcerpt.tokenRange;
+  }
+
+  /**
+   * Returns the `@typeParam` documentation for this parameter, if present.
+   */
+  public get tsdocTypeParamBlock(): tsdoc.DocParamBlock | undefined {
+    // TODO: TypeScript primarily supports the `@template` jsdoc tag, but tsdoc does not define it.
```

Comment:
TSDoc provides `@typeParam` for this.  See [here](https://github.com/microsoft/tsdoc/blob/40e313e3e54375b927240ed0cac2c770f863360f/tsdoc/src/details/StandardTags.ts#L326) and RFC https://github.com/microsoft/tsdoc/issues/72.

Reply:
I was speaking more to the fact that `@template` is recognized in the TypeScript compiler itself for JavaScript files.